app/models/concerns/has_custom_fields.rb

```diff
@@ -114,7 +114,7 @@ def self.preload_custom_fields(objects, fields)
 
         objects.each do |obj|
           map[obj.id] = obj
-          obj.preloaded_custom_fields = empty.dup
```

Comment:
Can we remove `attr_accessor :preloaded_custom_fields` now that we are setting the instance variable directly? 

Not sure if this is an official API but perhaps we should do it after the release if it isn't being used else where.

Reply:
Yes I agree, it should be `attr_reader`. It looks like it's not being used, `TopicList` does not use `HasCustomFields` and that's the only thing I see directly adding to `preloaded_custom_fields`:

![image](https://user-images.githubusercontent.com/920448/150464318-9cc6eb5c-0459-4067-bd01-9dceb1fd7a4f.png)

I will add a todo for post-release

Reply:
Ah it looks like we have places in the code base that relies on `attr_accessor`. Ignore this please :)

Reply:
Where are you seeing that @tgxworld ? I think it can be changed to `attr_reader` at least

Reply:
https://github.com/discourse/discourse/blob/c1ae214c7bb48f8baea7635dc165467cc65aa20d/app/serializers/basic_category_serializer.rb#L72

O but the line above is only reading the attribute. This change does feel a little risk on a Friday and with the release coming up on Monday though so I think we can revisit next week.

Reply:
Yeah no need to change the `attr_` thing before release anyway :) It can wait